nixos/modules/hardware/video/webcam/ipu6.nix

```
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.hardware.ipu6;
+
+  kernelPackages = config.boot.kernelPackages;
+
+in
+{
+
+  options.hardware.ipu6 = {
+
+    enable = mkEnableOption (lib.mdDoc "ipu6 kernel module");
+
+    platform = mkOption {
+      default = "ipu6";
+      type = types.enum [ "ipu6" "ipu6ep" ];
+      description = lib.mdDoc ''
+        Set platform to enable kernel module for.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    boot.extraModulePackages = [ kernelPackages.ipu6-drivers ];
+
+    hardware.firmware = optional (cfg.platform == "ipu6") pkgs.ipu6-camera-bin
+      ++ optional (cfg.platform == "ipu6ep") pkgs.ipu6ep-camera-bin;
```

```suggestion
    hardware.firmware = with pkgs; [ ]
      ++ optional (cfg.platform == "ipu6") ipu6-camera-bin
      ++ optional (cfg.platform == "ipu6ep") ipu6ep-camera-bin;
```

Maybe like this? Aligns both lines and makes it easier to see the difference between them.

---

done